### ISSUE TITLE
'Error: self signed certificate' even with domain in whitelist

### ISSUE THREAD

**Issue by oxchronxo** [httptoolkit/httptoolkit#297]
Today I rebooted my mac (Monterey 12.4 (21F79)) and fired up like normal; started VSCode, HTTPToolkit and Firefox(ProxyOmega). I loaded up my node project and got `Error: self signed certificate`. So I dug in

I'm working with a localhost using https://dev.blah.com:8088
I have /etc/hosts pointing my url to 127.0.0.1

All my system level CAs are up to date

I have HTTP2 support disabled for all clients(as I always do)

First I turned off ProxyOmega and tried a direct connection: PASS
Then I used `curl` to connect with no proxy: PASS
This I set up the proxy and tried to reconnect: FAIL
Then I used `curl -k`: FAIL

I checked the whitelist in HTTPToolkit and my host is still listed there. I removed it, restarted the proxy and added it back in.
Then I used `curl` to connect with the proxy: FAIL

Here is the output.
```
$ curl -vv -k https://dev.blah.com:8088
* Uses proxy env variable https_proxy == 'http://127.0.0.1:8000'
*   Trying 127.0.0.1:8000...
* Connected to 127.0.0.1 (127.0.0.1) port 8000 (#0)
* allocate connect buffer!
* Establish HTTP proxy tunnel to dev.blah.com:8088
> CONNECT dev.blah.com:8088 HTTP/1.1
> Host: dev.blah.com:8088
> User-Agent: curl/7.79.1
> Proxy-Connection: Keep-Alive
> 
< HTTP/1.1 200 OK
< 
* Proxy replied 200 to CONNECT request
* CONNECT phase completed!
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (OUT), TLS handshake, Client hello (1):
* (304) (IN), TLS handshake, Server hello (2):
* (304) (IN), TLS handshake, Unknown (8):
* (304) (IN), TLS handshake, Certificate (11):
* (304) (IN), TLS handshake, CERT verify (15):
* (304) (IN), TLS handshake, Finished (20):
* (304) (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / AEAD-AES256-GCM-SHA384
* ALPN, server accepted to use http/1.1
* Server certificate:
*  subject: CN=dev.blah.com; O=Mockttp Cert - DO NOT TRUST
*  start date: Jun 24 20:22:21 2022 GMT
*  expire date: Jun 25 20:22:21 2023 GMT
*  issuer: CN=HTTP Toolkit CA
*  SSL certificate verify ok.
> GET / HTTP/1.1
> Host: dev.blah.com:8088
> User-Agent: curl/7.79.1
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 502 Error communicating with upstream server
* no chunk, no close, no size. Assume close to signal end
< 
* Closing connection 0
Error: self signed certificate
```

Any help would be appreciated

---

_Does this affect you too? Click the +:smiley: icon in the top right and add a :+1: on issues to help decide where HTTP Toolkit goes next, or [go take a look & vote on the most popular ideas so far](https://github.com/httptoolkit/feedback/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc)._

**Comment by pimterry** [comment]
Hi @oxchronxo. Thanks for reporting this. This was due to a small bug after upgrading to Mockttp v3 - I've now deployed a fix for that. HTTP Toolkit will update in the background next time you run the app (just wait 30 seconds or so) and then the next time everything should work correctly.

Sorry about that, thank you for reporting this! Everything should be resolved now, but do let me know if you're still having any issues.
